src/motor_hardware.cc

```diff
@@ -142,6 +144,27 @@ void MotorHardware::readInputs() {
                     }
                     break;
                 }
+                case MotorMessage::REG_BATTERY_VOLTAGE: {
+                    int32_t data = mm.getData();
+                    sensor_msgs::BatteryState bstate;
+                    bstate.voltage = (float)data * pid_params.battery_voltage_multiplier +
+                                   pid_params.battery_voltage_offset;
+                    bstate.current = std::numeric_limits<float>::quiet_NaN();
+                    bstate.charge = std::numeric_limits<float>::quiet_NaN();
+                    bstate.capacity = std::numeric_limits<float>::quiet_NaN();
+                    bstate.design_capacity = std::numeric_limits<float>::quiet_NaN();
+                    if (bstate.voltage < 22.0) { // 18V = 0%,  22V = 12%
+                        bstate.percentage = std::max(0.0, 2.0 * (bstate.voltage - 18.0));
```

Comment:
`bstate.percentage` should be between 0 and 1. `0.02 * (bstate.voltage - 18.0)` should make it work

Reply:
Also I think our lower voltage (0% level) should be 20V. at 18 you are at 9V per battery which is pretty low.